[sandyk0105] iP

sandyk0105 opened the pull request:
# REI
> “Don't put off until tomorrow what you can do today.” - Benjamin Franklin
## Description
**REI** is a simple yet promising chatbot, inspired by the charm and creativity of a member from the South Korean girl group [IVE :cupid:](https://kprofiles.com/ive-members-profile/). Though currently in its early stages, **REI** is designed to grow and evolve into a highly useful and intelligent chat companion. 

For now, **REI** helps users by efficiently storing tasks, but with time, it will develop into a more versatile assistant, adding more features and improving with every interaction. There are three types of task **REI** can handle
1. *ToDo*: A simple, regular task without any specific time constraints. Great for quick reminders or ongoing activities.
2. *Deadline* : A task that needs to be completed by a specific date or time. Perfect for time-sensitive assignments or projects.
3. *Event* : A task with a defined start and finish time, making it ideal for meetings, appointments, or scheduled activities.

**REI** brings a fresh K-pop-inspired twist to task management, making everyday tasks a bit more fun!

## Features
- Create a task of a specific type (*ToDo*/*Deadline*/*Event*) | `todo`, `deadline`, `event`
- Delete a ~task~ | `delete`
- Mark a task as done | `mark`
- Unmark a task as done | `unmark`
- Display the list of tasks | `list`
- Find a specific task using keywords | `find`


## Usage Example
```
-----------YOU------------
list
-----------REI♥-----------
Here are the tasks in your list: 
1. [T][ ] read book
2. [D][X] return book (by: Mon, Dec 2 2019 15:00:00)
3. [D][ ] project meeting (by: Mon, Dec 2 2019 12:00:00)
4. [E][ ] meeting (from: Mon, Dec 2 2019 12:00:00 to: Wed, Jan 1 2020 10:00:00)
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

sandyk0105 committed:
Level 0 : Bot Rename

I renamed my bot to "Rei"

sandyk0105 committed:
Level 1

Customized Rei bot

sandyk0105 committed:
Level 2 : List

Updated Rei to return list of tasks

sandyk0105 committed:
Level-3: Mark as done/not done

Updated rei to mark tasks as done/not done. Rei can also handle task numbers beyond limit (ask user to reinput the correct task number)

sandyk0105 committed:
Level 4 : ToDo, Deadline, Event

Made ToDo, Deadline, and Event inner classes of Task class.

sandyk0105 committed:
A-TextUiTesting

Make testcases and desired outputs

sandyk0105 committed:
Level 5 : Handle errors

Handle errors for unknown keywords and user not specifying details

sandyk0105 committed:
Level-6 : Delete

Allow user to delete task. Use ArrayList to store the tasks.

Meowloid commented on src/main/java/Rei.java:
```diff
@@ -0,0 +1,168 @@
+
+        System.out.println("Annyeong! I'm\n" + logo);
+        System.out.println("What can I do for you?");
+        System.out.println("-----------YOU------------");
```
I like the creativity here: the printlines demarcate clearly who's printing what, like a chat log.

Meowloid commented on src/main/java/Rei.java:
```diff
@@ -0,0 +1,168 @@
+import java.util.Scanner;
+import java.util.List;
+
+public class Rei {
```
I think it'd be good to have a method/static constant to represent line breaks and other things. I put mine in the Ui class cause I think it makes the most sense, but anywhere's fine, probably. Here's what I mean
```
public void printLine() {
        System.out.println("____________________________________________________________");
    }
```

Meowloid commented on src/main/java/Rei.java:
```diff
@@ -0,0 +1,168 @@
+                }
+            } else if (scanner.hasNext("todo")
+                        || scanner.hasNext("deadline")
+                        || scanner.hasNext("event")) {
```
I like this grouping of commands: it makes more sense especially when you go deeper down the OOP rabbithole and have many classes representing different tasks. This inspires me to incorporate this into my code.

kikokidayo replied:
Overall most method and variable names (eg. `isAllWhitespace`) are quite clear and descriptive, nice!

kikokidayo replied:
Just a small nitpick but could consider changing `id` to something a bit clearer like `listIndex`?

Meowloid reviewed (commented):
LGTM. Promising. I hope to see you go the full stretch and incorporate the weekly tasks. I did not have fun refactoring my code either, but I understand the need for it. I'm sure you'll be able to get inspiration from doing the PR reviews from others' code.

kikokidayo commented on src/main/java/Rei.java:
Should consider adding Javadoc comments for the `Rei` class and maybe some important methods as well

kikokidayo commented on src/main/java/Task.java:
Could consider adding JavaDoc comment for `Task` class as well.

kikokidayo replied:
Consider putting the `Deadline` class in its own separate file for more organisation?

kikokidayo reviewed (commented):
Overall LGTM! Can just consider adding more Javadoc comments overall

sandyk0105 committed:
Level 7 : Read database

Allow rei bot to store tasks in a database.

sandyk0105 committed:
Level 8: Date time

Allow rei bot to take and use date-time format.

sandyk0105 committed:
A-MoreOOP

Extract Parser, Storage, TaskList and Ui from Rei

sandyk0105 committed:
Merge branch 'add-gradle-support'

sandyk0105 committed:
A-JUnit

Make JUnit testcases for TaskList and Task

sandyk0105 committed:
A-Jar

Package Rei bot into JAR file using Gradle shadowJar

sandyk0105 committed:
Extract method

Add method in TaskList that returns the storing format

sandyk0105 committed:
Parser update

Add enums in parser for better abstraction

sandyk0105 committed:
Delete Ui instance

Delete Ui instance in Rei

sandyk0105 committed:
add JavaDoc

Add JavaDoc to all classes

sandyk0105 committed:
Follow Coding Standard

sandyk0105 committed:
Add Find method

Add find method using a keyword

sandyk0105 committed:
Not changing parameter

Delete the part where a parameter's value is changed

sandyk0105 committed:
Merge branch 'branch-A-CodingStandard'

sandyk0105 committed:
Merge branch 'branch-Level-9'

sandyk0105 committed:
Update Save

Save the tasks directly (does not wait for an exit command)

sandyk0105 committed:
Level 10 : GUI

Create GUI for REI using FXML

Modify the methods to support the GUI

sandyk0105 committed:
Clean up code

Standardize error handling with ReiException and remove unused code

The project still contains some non-GUI error handling which doesn't fit
well with the current GUI-based structure. There are also several unused
methods in TaskList and unused fields across the codebase. Additionally,
the Ui.java class has become obsolete since the transition to a full GUI
interface.

Non-GUI error handling fragments the code and leads to inconsistent
behavior in certain parts of the application. Unused code increases
complexity and can cause confusion for developers.

Let's replace all non-GUI error handling with ReiException to
standardize how exceptions are handled. This will ensure consistent
error reporting within the GUI.

Let's also clean up the codebase by removing unused methods from
TaskList and other unused fields. These removals will reduce technical
debt and improve maintainability.

Finally, we'll delete Ui.java as it is no longer required due to the
complete shift to GUI-based user interaction. This cleans up the codebase
further and reflects the current architecture of the project.

sandyk0105 committed:
Update javadocs for TaskList message changes

The current javadocs are outdated and do not describe the changes
made to the TaskList methods, which now generate messages upon
executing commands.

This needs to change because the javadocs should accurately
document the behavior of the methods, especially to ensure clarity
for future contributors and maintainers.

Update the javadocs to describe the message generation after
command execution.

This is done to keep the documentation aligned with the current
implementation, making it easier to understand the methods'
functionality.

No functional code changes have been made—this commit strictly
updates the documentation.

sandyk0105 committed:
Add assertion

Add assertion in the defauls case of the enum switch-case

sandyk0105 committed:
Refactor: apply SLAP in Parser.java

The current codebase has sections that do not adhere to the
SLAP, making it harder to read and maintain.

This needs to change to improve code readability and ensure
a cleaner structure by keeping each method focused
on a single level of abstraction.

Extract the lines of code that violate SLAP into separate methods.

Tested the code to verify that it still works the same after the
refactor.

sandyk0105 committed:
Fix: correct delete command to handle task index properly

The delete command was not functioning as expected due to the
taskIndex being incorrectly parsed. The input was not trimmed,
leading to errors when extracting the number from the prompt.

This fix trims the input before passing it to `parseInt`, ensuring
the correct task index is extracted.

This resolves the issue, allowing the delete command to work
as intended.

sandyk0105 committed:
Merge pull request #2 from sandyk0105/branch-A-Assertions

Add an assertion

sandyk0105 committed:
Merge branch 'master' of https://github.com/sandyk0105/ip

sandyk0105 committed:
Merge pull request #3 from sandyk0105/master

Sync branch-A-CodeQuality with the latest master

sandyk0105 committed:
Give credit to inspiration author

Add proper credit to the author who inspired this piece of idea

sandyk0105 committed:
Merge pull request #4 from sandyk0105/branch-A-CodeQuality

Improve Code Quality by Fixing SLAP

sandyk0105 committed:
Implement tagging system for tasks

- add ability to assign a tag or a list of tags to a Task
- implement functionality to load and save tags

sandyk0105 committed:
Implement tag command and enhance find command

Add tag command to allow tagging tasks

Modify find command to support finding tasks by tag

sandyk0105 committed:
Implement untag command

Implemented untag command to remove tags from tasks

sandyk0105 committed:
Implement viewtags command

Add viewtags command to display tags of a tas

sandyk0105 committed:
Merge branch 'branch-C-Tagging' into master

sandyk0105 committed:
Merge pull request #5 from sandyk0105/master

Bring CodeQuality changes to the branch-C-Tagging

sandyk0105 committed:
Merge pull request #6 from sandyk0105/branch-C-Tagging

Merge branch-C-Tagging safely

sandyk0105 committed:
Improve code quality by applying more SLAPs

Extract tag-related methods in Rei.java, aligning them with the
structure of existing commands .

Add an assertion in Storage::load to ensure Task types during
the loading process.

sandyk0105 committed:
Merge pull request #7 from sandyk0105/branch-C-Tagging

Improve code quality by applying more SLAPs

sandyk0105 committed:
Minimizing the use of magic numbers

Extract magic numbers in Rei.java
into constants

sandyk0105 committed:
Merge branch 'master' of https://github.com/sandyk0105/ip

sandyk0105 committed:
Profile update

Change user and bot profile pictures

sandyk0105 committed:
Update viewtags to handle no tags

Handle the case when viewtag
is called on a Task with no tags

sandyk0105 committed:
GUI Enhancements

Design GUI enhancements:
- Remove borders from message bubbles
- Flip message bubble for the bot
- Update color palette for improved aesthetics
- Change profile pictures to circular shapes

sandyk0105 committed:
Merge pull request #8 from sandyk0105/branch-Better-GUI

GUI Enhancements

sandyk0105 committed:
Create User Guide

Edit README to fit the user guide
following the GFMD

sandyk0105 committed:
Refactor storage system: switch from Path to File to resolve loading issues

The storage system currently uses Path for file
handling, which fails when loading new files.

This needs to change because the failure to load new files indicates
compatibility issues and limitations in file operations.

Adapted from github.com/adipanda2002/ip .

sandyk0105 committed:
omplete Javadocs for DialogBox.java, Main.java, and MainWindow.java

Complete Javadocs for DialogBox.java, Main.java, and MainWindow.java

The lack of Javadocs in DialogBox.java, Main.java, and
MainWindow.java makes the code harder to understand and maintain.

This can cause confusion for developers working on or extending the
codebase.

Let’s add Javadocs to explain the purpose of key methods and
constructors, including their parameters and behavior.

This will improve readability, aid future developers, and make the
codebase easier to maintain and collaborate on.

sandyk0105 committed:
Add Javadocs to methods in Parser.java

The methods in Parser.java lack documentation, making it hard for
developers to understand their purpose and usage, especially since
the input parameters are strings.

This absence of Javadocs can lead to confusion for anyone working
with these methods, hindering effective collaboration and future
development.

Let’s add Javadocs for the following methods that take a string as
input:

getUnmarkPrompt(prompt)
getToDoPrompt(prompt)
getDeadlinePrompt(prompt)
getEventPrompt(prompt)
getDeletePrompt(prompt)
getFindPrompt(prompt)
getTagPrompt(prompt)
getUntagPrompt(prompt)
getViewtagsPrompt(prompt)

Adding these Javadocs enhances code clarity.

sandyk0105 committed:
Revert "omplete Javadocs for DialogBox.java, Main.java, and MainWindow.java"

This reverts commit 38c663a37c6430efa8d2f8e7c2cff001f4c8c5c9.

sandyk0105 committed:
Complete Javadocs for DialogBox.java, Main.java, and MainWindow.java

Complete Javadocs for DialogBox.java, Main.java, and MainWindow.java

The lack of Javadocs in DialogBox.java, Main.java, and
MainWindow.java makes the code harder to understand and maintain.

This can cause confusion for developers working on or extending the
codebase.

Let’s add Javadocs to explain the purpose of key methods and
constructors, including their parameters and behavior.

This will improve readability, aid future developers, and make the
codebase easier to maintain and collaborate on.

sandyk0105 committed:
Clean up imports and improve documentation

Clean up imports and improve documentation

Unused imports in Rei.java clutter the code and can lead to confusion.

Cleaning up these imports will enhance code readability and maintainability.

Let’s remove unused imports in Rei.java, specify the imports used in
Storage.java instead of using an asterisk, and fix a typo in
Storage.java. Additionally, Javadocs will be added for the
ReiException.java class to improve clarity and documentation.

These changes contribute to a cleaner codebase and better
documentation, making it easier for developers to understand and
work with the code.

sandyk0105 committed:
Remove Unused imports in Task.java

Remove Unused Imports in Task.java

Unused imports in Task.java clutter the code and can create
confusion.

Removing these imports enhances code readability and keeps the
codebase clean.
